### DISCUSSION TITLE
功能请求：通过xesam:asText 获取歌词

### DISCUSSION THREAD

**Discussion by yigekuyou** [KangweiZhu/lyrics-on-panel#10] [Ideas]
一些播放器已经支持此项
[go-musicfox](https://github.com/go-musicfox/go-musicfox/issues/202)
[netease-cloud-music-gtk](https://github.com/gmg137/netease-cloud-music-gtk/issues/268) 
通过播放器提供的歌词比网络下载更加正确

**Comment by KangweiZhu** [comment]
好主意。等我有空试下。kde5应该是能纯qml实现的。正好到时候顺手把kde5版本的代码重构下。

kde6就大概率得额外做个后端了。单纯依赖qml不太可能拿到这个信息。因为kde6 mpris相关的qml api有点半成品的感觉，很多东西都名存实亡。

**Comment by KangweiZhu** [comment]
抱歉, 回国做.. 这俩播放器音源得配代理。

**Reply by yigekuyou** [reply]
我写了个cpp 的 qml https://github.com/yigekuyou/lyrics-on-panel/tree/test

**Reply by yigekuyou** [reply]
问题又来到了 
歌词 是这样的 
[lyric.irc.tar.gz](https://github.com/user-attachments/files/22397278/lyric.irc.tar.gz)
